src/System.Devices.Gpio/System.Devices.Gpio.Samples/RgbColorSensor.cs

```diff
@@ -0,0 +1,466 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Devices.I2c;
+using System.Drawing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Devices.Gpio.Samples
+{
+    public enum RgbColorSensorIntegrationTime : byte
+    {
+        Milliseconds2_4 = 0xFF,  // 2.4ms -  1 cycle   - Max Count: 1024
+        Milliseconds24 = 0xF6,   //  24ms - 10 cycles  - Max Count: 10240
+        Milliseconds50 = 0xEB,   //  50ms - 20 cycles  - Max Count: 20480
+        Milliseconds101 = 0xD5,  // 101ms - 42 cycles  - Max Count: 43008
+        Milliseconds154 = 0xC0,  // 154ms - 64 cycles  - Max Count: 65535
+        Milliseconds700 = 0x00   // 700ms - 256 cycles - Max Count: 65535
+    }
+
+    public enum RgbColorSensorGain : byte
+    {
+        Gain1X = 0x00,   //  No gain
+        Gain4X = 0x01,   //  2x gain
+        Gain16X = 0x02,  //  16x gain
+        Gain60X = 0x03   //  60x gain
+    }
+
+    /// <summary>
+    /// Supports TCS34725 RGB Color sensor
+    /// </summary>
+    public class RgbColorSensor : IDisposable
+    {
+        private const byte DEVICE_ADDRESS = 0x29; // Default I2C address
```

Comment:
It looks like variations of this device also use 0x39

https://cdn-shop.adafruit.com/datasheets/TCS34725.pdf (see page 3); is it worth making this more general purpose?